jazz_delete-serverless-service/index.js

```diff
@@ -28,141 +28,145 @@ var payloads = formats('apis');
-        logger.debug("Started sendEventsOptions:" + JSON.stringify(sendEventsOptions));
-        sendEventToHandler(sendEventsOptions);
-        request(req, function (error, response, body) {
-            if (error) {
-                logger.error("request errored..: " + JSON.stringify(error));
-                sendEventsOptions.error = "Failed..: " + JSON.stringify(error);
-                sendEventToHandler(sendEventsOptions);
-                return cb(JSON.stringify(errorHandler.throwInternalServerError("Internal error occurred")));
-            } else {
-                if (response.statusCode === 200 || response.statusCode === 201) {
-                    logger.info("success..: " + JSON.stringify(response));
-                    payloads.responseLoad.request_id = tracking_id;
-                    sendEventsOptions.event_status = "COMPLETED";
-                    //Send COMPLETED events for CALL_DELETE_WORKFLOW/CALL_DELETE_ENV_WORKFLOW
-                    logger.debug("Completed sendEventsOptions:" + JSON.stringify(sendEventsOptions));
-                    sendEventToHandler(sendEventsOptions);
-                    return cb(null, responseObj(payloads.responseLoad, event.body));
-                } else {
-                    //Send FAILED event for CALL_DELETE_WORKFLOW/CALL_DELETE_ENV_WORKFLOW
-                    var errMessage = "Internal error occurred";
-                    sendEventsOptions.error = "Failed..: " + JSON.stringify(response);
-                    sendEventsOptions.event_status = "FAILED";
-                    if (response.statusCode === 401) {
-                        errMessage = "Not authorized";
-                    }
-                    logger.error("Failed..: " + JSON.stringify(response));
-                    logger.debug("Failed sendEventsOptions:" + JSON.stringify(sendEventsOptions));
-                    sendEventToHandler(sendEventsOptions);
-                    return cb(JSON.stringify(errorHandler.throwInternalServerError(errMessage)));
-                }
-            }
-        });
-
+      });
     } catch (ex) {
-        //Send FAILED event for CALL_DELETE_WORKFLOW/CALL_DELETE_ENV_WORKFLOW
-        sendEventsOptions.error = "Failed..: " + JSON.stringify(ex.message);
-        sendEventsOptions.event_status = "FAILED";
-        sendEventToHandler(sendEventsOptions);
-        logger.error('Error : ', ex.message);
-        cb(JSON.stringify(errorHandler.throwInternalServerError("Internal error occurred")));
-    }
-
-    function sendEventToHandler(sendEventsOptions) {
-        //Send Started/Completed/Failed events for CALL_DELETE_WORKFLOW /CALL_DELETE_ENV_WORKFLOW
-        eventHandler.sendEvent(sendEventsOptions, function (err, data) {
-            if (err) {
-                logger.error(err.message);
-            } else {
-                logger.info(data.message);
-            }
-        });
+      logger.error("Failed to send event: " + JSON.stringify(ex));
```

Comment:
We might need to be consistent with these how we are logging errors/messages. JSON.stringify(ex) might not work with all exceptions. Might need to log ex object or ex.message?

Reply:
AFAIK, this would work (including the exception object). The corner cases are when there are circular references in which case JSON.stringify will detect it and just spit out [object] or some variation of it.